### PR TITLE
Remove never-supported DOMConfiguration/DOMImplementationList

### PR DESCRIPTION
**Pull request by saschanaz**
A checklist to help your pull request get merged faster:
- [x] Summarize your changes
- [ ] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)
- [ ] Data: if you tested something, describe how you tested with details like browser and version
- [ ] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [x] Link to related issues or pull requests, if any: #6854, #6768

### CODE REVIEW

File: api/DOMConfiguration.json [hunk 1]
```diff
@@ -1,244 +0,0 @@
-{
-  "api": {
-    "DOMConfiguration": {
-      "__compat": {
-        "mdn_url": "https://developer.mozilla.org/docs/Web/API/DOMConfiguration",
```
Comment (foolip) [inline api/DOMConfiguration.json:5]: This went together with `document.domConfig`, which was once in Firefox:
https://github.com/mdn/browser-compat-data/blob/c1f99a1ee3166e343cd509d7e205a06f890fdc46/api/Document.json#L3525-L3532

But it was so long ago now that the entry can still be removed. Can you roll that into this change a well?

File: api/DOMImplementationList.json [hunk 2]
```diff
@@ -1,148 +0,0 @@
-{
-  "api": {
-    "DOMImplementationList": {
```
Comment (foolip) [inline api/DOMImplementationList.json:3]: This went together with https://www.w3.org/TR/DOM-Level-3-Core/core.html#DOMImplementationSource, which isn't in BCD. This removal seems appropriate.

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
Over to @chrisdavidmills . This pages should be deleted:
https://developer.mozilla.org/en-US/docs/Web/API/DOMConfiguration
https://developer.mozilla.org/en-US/docs/Web/API/DOMImplementationList

These pages are 404s and no action needed:
https://developer.mozilla.org/en-US/docs/Web/API/DOMConfiguration/canSetParameter
https://developer.mozilla.org/en-US/docs/Web/API/DOMConfiguration/getParameter
https://developer.mozilla.org/en-US/docs/Web/API/DOMConfiguration/parameterNames
https://developer.mozilla.org/en-US/docs/Web/API/DOMConfiguration/setParameter
https://developer.mozilla.org/en-US/docs/Web/API/DOMImplementationList/item
https://developer.mozilla.org/en-US/docs/Web/API/DOMImplementationList/length

**Comment by chrisdavidmills** [comment]
@foolip 

* https://wiki.developer.mozilla.org/en-US/docs/Web/API/Document/domConfig redirected to its parent element.
* https://developer.mozilla.org/en-US/docs/Web/API/DOMConfiguration deleted.
* https://developer.mozilla.org/en-US/docs/Web/API/DOMImplementationList deleted.

**Pull request merged**